swaybar/render.c

```diff
@@ -286,59 +286,67 @@ void render(struct output *output, struct config *config, struct status_line *li
 	cairo_set_operator(cairo, CAIRO_OPERATOR_CLEAR);
 	cairo_paint(cairo);
 	cairo_restore(cairo);
-
-	cairo_set_operator(cairo, CAIRO_OPERATOR_SOURCE);
-
-	// Background
-	if (is_focused) {
-		cairo_set_source_u32(cairo, config->colors.focused_background);
-	} else {
-		cairo_set_source_u32(cairo, config->colors.background);
-	}
-	cairo_paint(cairo);
-
-	// Command output
-	if (is_focused) {
-		cairo_set_source_u32(cairo, config->colors.focused_statusline);
-	} else {
-		cairo_set_source_u32(cairo, config->colors.statusline);
-	}
-
-	int width, height;
-
-	if (line->protocol == TEXT) {
-		get_text_size(window->cairo, window->font, &width, &height,
-				window->scale, config->pango_markup, "%s", line->text_line);
-		cairo_move_to(cairo, (window->width * window->scale)
-				- margin - width, margin);
-		pango_printf(window->cairo, window->font, window->scale,
-				config->pango_markup, "%s", line->text_line);
-	} else if (line->protocol == I3BAR && line->block_line) {
-		double pos = (window->width * window->scale) - 0.5;
-		bool edge = true;
-		for (i = line->block_line->length - 1; i >= 0; --i) {
-			struct status_block *block = line->block_line->items[i];
-			if (block->full_text && block->full_text[0]) {
-				render_block(window, config, block, &pos, edge, is_focused);
-				edge = false;
+	// Could also be done by always rendering then conditionally clearing and drawing alpha
+	// That may be preferable down the line
+	if ((!strcmp(config->display_mode, "hide")) == 0 || strcmp(config->hidden_state, "show") == 0) {
```

Comment:
No - in display_mode "hide" when the modifier is held down, hidden_state changes to "show", so the first clause will evaluate to false, while the second evaluates to true.
We could (and potentially should) make it so that we only need the second clause. I added the first because I think hidden_state defaults to "hide" but we want the bar to show anyway as long as the display_mode isn't "hide"